File: app/src/main/java/pl/srw/billcalculator/BackableActivity.java
package pl.srw.billcalculator;

import android.os.Bundle;
import android.view.MenuItem;

import pl.srw.mfvp.MvpActivity;
import pl.srw.mfvp.di.MvpComponent;
import timber.log.Timber;

public abstract class BackableActivity<T extends MvpComponent> extends MvpActivity<T> {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getSupportActionBar() != null)
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onBackPressed() {
        Timber.i("Back pressed");
        super.onBackPressed();
    }
}

File: app/src/main/java/pl/srw/billcalculator/common/BackableActivity.kt
package pl.srw.billcalculator.common

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import timber.log.Timber

abstract class BackableActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        Timber.i("Back pressed")
        super.onBackPressed()
    }
}
